Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindowManager.showBootMessage:COMMENT
Method Modifier: public      internal    
Comment:/**
 * {@inheritDoc}
 */

@@ -1,12 +1,18 @@
 {
-    if (mHeadless)
-        return;
     mHandler.post(new Runnable() {
 
         @Override
         public void run() {
             if (mBootMsgDialog == null) {
-                mBootMsgDialog = new ProgressDialog(mContext) {
+                int theme;
+                if (mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_WATCH)) {
+                    theme = com.android.internal.R.style.Theme_Micro_Dialog_Alert;
+                } else if (mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEVISION)) {
+                    theme = com.android.internal.R.style.Theme_Leanback_Dialog_Alert;
+                } else {
+                    theme = 0;
+                }
+                mBootMsgDialog = new ProgressDialog(mContext, theme) {
 
                     // This dialog will consume all events coming in to
                     // it, to avoid it trying to do things too early in boot.

